pkg/aws/client.go

```diff
@@ -423,32 +403,45 @@ func (b *ClientBuilder) Build() (Client, error) {
 	return c, err
 }
 
-func (c *awsClient) GetIAMCredentials() (credentials.Value, error) {
-	return c.awsSession.Config.Credentials.Get()
+func (c *awsClient) GetIAMCredentials() (aws.Credentials, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return aws.Credentials{}, err
+	}
+
+	creds, err := cfg.Credentials.Retrieve(context.TODO())
+	if err != nil {
+		return aws.Credentials{}, err
+	}
+	return creds, nil
 }
 
-func (c *awsClient) GetRegion() string {
-	return aws.StringValue(c.awsSession.Config.Region)
+func (c *awsClient) GetRegion() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
```

Comment:
Previously we stored the `awsSession` in the `awsClient` struct, and accessed it to fetch the region.
Since the `session` isn't available with V2, should we store the `cfg` in the `awsClient`, so we can access here `cfg.Region` instead of loading the config, that way we also don't need to handle an error here.